usaspending_api/agency/v2/views/subcomponents.py

```diff
@@ -0,0 +1,85 @@
+from django.db.models import Q, Sum, OuterRef, Subquery, F, Value
+from rest_framework.request import Request
+from rest_framework.response import Response
+from typing import Any
+from usaspending_api.agency.v2.views.agency_base import AgencyBase, PaginationMixin
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
+from usaspending_api.common.helpers.orm_helpers import ConcatAll
+from usaspending_api.references.models import GTASSF133Balances, BureauTitleLookup
+
+
+class SubcomponentList(PaginationMixin, AgencyBase):
+    """
+    Obtain the count of subcomponents (bureaus) for a specific agency in a single
+    fiscal year based on GTAS
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/toptier_code/sub_components.md"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.params_to_validate = ["toptier_code", "fiscal_year"]
+
+    @cache_response()
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        self.sortable_columns = ["name", "total_obligations", "total_outlays", "total_budgetary_resouces"]
+        self.default_sort_column = "total_obligations"
+        results = self.format_results(self.get_subcomponents_queryset())
+        page_metadata = get_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        return Response(
+            {
+                "toptier_code": self.toptier_code,
+                "fiscal_year": self.fiscal_year,
+                "results": results[self.pagination.lower_limit : self.pagination.upper_limit],
+                "messages": self.standard_response_messages,
+                "page_metadata": page_metadata,
+            }
+        )
+
+    def format_results(self, response):
+        results = sorted(
+            [
+                {
+                    "name": x["bureau_info"].split(";")[0],
```

Comment:
Not sure the chances of this occurring in Production data; however, should handle the case of a NoneType here. Local doesn't have data in it and throws an error because of `None.split()`.